### PR TITLE
Refactoring Tests & Various Fixes for Symfony 6

### LINKED ISSUE #900: [make:registration] No Guard authenticators found - New Authenticator
Hi,
I have Symfony 5.3.2.
I do : `php bin/console make:registration-form`.

I answered question 'yes' for this question `Do you want to automatically authenticate the user after registration? (yes/no) [yes]:`.

And I have ` ! [NOTE] No Guard authenticators found - so your user won't be automatically authenticated after registering.`.

### LINKED ISSUE #984: Prep Security (make:user) for 6.0 release
Currently, for Symfony 5.x, we still generate (in `make:user`) deprecated methods (because we need to). For example, if I run `make:user` but choose NO to needing a password, it generates:

```php
// ...
class User implements UserInterface
{
    // ...

    /**
     * This method can be removed in Symfony 6.0 - is not needed for apps that do not check user passwords.
     *
     * @see PasswordAuthenticatedUserInterface
     */
    public function getPassword(): ?string
    {
        return null;
    }
}
```

We need to prep MakerBundle before the 6.0 release to be smart enough to *not* generate these anymore (if you're using 6.0).

### LINKED ISSUE #1005: make:user password yes
### Symfony version(s) affected

5.3.10

### Description

make:user password yes

In SecurityConfigUpdater.php line 188:

  Symfony\Bundle\MakerBundle\Security\SecurityConfigUpdater::updatePasswordHashers(): Argument symfony/symfony#1 ($userConfig) must
  be of type Symfony\Bundle\MakerBundle\Security\UserClassConfiguration, string given, called in /var/www/vendor/symf
  ony/maker-bundle/src/Security/SecurityConfigUpdater.php on line 193

### How to reproduce

-

### Possible Solution

_No response_

### Additional Context

_No response_

### LINKED ISSUE #1009: Controllers shouldn't use getDoctrine
Per https://symfony.com/blog/new-in-symfony-5-4-controller-changes, this generates deprecated code:

```
$ grep getDoctrine  src/ -r
src/Resources/skeleton/crud/controller/Controller.tpl.php:        $<?= $entity_var_plural ?> = $this->getDoctrine()
src/Resources/skeleton/crud/controller/Controller.tpl.php:            $entityManager = $this->getDoctrine()->getManager();
src/Resources/skeleton/crud/controller/Controller.tpl.php:            $this->getDoctrine()->getManager()->flush();
src/Resources/skeleton/crud/controller/Controller.tpl.php:            $entityManager = $this->getDoctrine()->getManager();
src/Resources/skeleton/registration/RegistrationController.tpl.php:            $entityManager = $this->getDoctrine()->getManager();
src/Resources/skeleton/resetPassword/ResetPasswordController.tpl.php:            $this->getDoctrine()->getManager()->flush();
src/Resources/skeleton/resetPassword/ResetPasswordController.tpl.php:        $user = $this->getDoctrine()->getRepository(<?= $user_class_name ?>::class)->findOneBy([
```

### PR DESCRIPTION
**Pull request by weaverryan**
Hi!

A big, kind of ugly change. Reasons:

* We needed a bit more flexibility to tweak the workflow of individual tests for different versions of Symfony
* Reduced duplication: the "fixtures" in the tests caused a lot of duplication, which has grown over time. This minimizes that quite a lot.

Fixes #1009, fixes #1005, fixes #984, fixes #900

Cheers!

### CODE REVIEW

File: src/Maker/MakeRegistrationForm.php [hunk 1]
```diff
@@ -328,6 +339,7 @@ public function generate(InputInterface $input, ConsoleStyle $io, Generator $gen
                     'email_getter' => $this->emailGetter,
                     'authenticator_class_name' => $this->autoLoginAuthenticator ? Str::getShortClassName($this->autoLoginAuthenticator) : null,
                     'authenticator_full_class_name' => $this->autoLoginAuthenticator,
+                    'uses_authenticator_system' => $this->useSecurity52,
```
Comment (OskarStark) [inline src/Maker/MakeRegistrationForm.php:342]:
```suggestion
                    'use_new_authenticator_system' => $this->useNewAuthenticatorSystem,
```
Looks clearer to me, if needed we can add 52 suffix to the variable name

File: src/Test/MakerTestRunner.php [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+            'postgresql://symfony:ChangeMe@127.0.0.1:5432/app?serverVersion=13&charset=utf8',
+            getenv('TEST_DATABASE_DSN')
+        );
+
+        // Flex includes a recipe to suffix the dbname w/ "_test" - lets keep
+        // things simple for these tests and not do that.
+        $this->replaceInFile(
+            'config/packages/test/doctrine.yaml',
+            "dbname_suffix: '_test%env(default::TEST_TOKEN)%'",
+            '')
```
Comment (OskarStark) [inline src/Test/MakerTestRunner.php:142]: ) 

to the next line

File: src/Test/MakerTestRunner.php [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+    {
+        $this->replaceInFile(
+            '.env',
+            'postgresql://symfony:ChangeMe@127.0.0.1:5432/app?serverVersion=13&charset=utf8',
+            getenv('TEST_DATABASE_DSN')
+        );
+
+        // Flex includes a recipe to suffix the dbname w/ "_test" - lets keep
+        // things simple for these tests and not do that.
+        $this->replaceInFile(
```
Comment (OskarStark) [inline src/Test/MakerTestRunner.php:139]: Would be nice to have removeFromFile() or removeInFile() method which does the replacement internally.

It's easier to read

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -82,28 +82,28 @@ jobs:
...
             matrix:
                 php-version:
-                    - '7.1.3'
-                    - '7.2'
-                    - '7.3'
-                    - '7.4'
                     - '8.0'
-                symfony-skeleton-stability:
-                    - 'stable'
+                symfony-version:
```
[the remark this answers is not in the document]
Comment (weaverryan) [reply .github/workflows/ci.yaml:93]: I'm not sure what you mean by this one? This should be under matrix, right?

[the remark this answers is not in the document]
Comment (OskarStark) [reply .github/workflows/ci.yaml:93]: damn sorry, the indention in the Github Mobile app was different oO

**Comment by weaverryan** [comment]
This is ready! Though I don't seriously expect any detailed review :).

This is HUGE - sorry. It is 95% test refactoring and 3% fixing our test matrix to test a wider variety of Symfony versions and 2% fixing minor real bugs for different versions of Symfony that these enhanced matrix found.

The remaining failures for Symfony 6 are due to 10 tests where the package we're testing is not yet installable on Symfony 6. So, not much we can do there :). Those tests should start passing when those packages add Sf6 compat. If they continue failing, we'll then know we have a bug to track down.

Cheers!

**Pull request merged**